src/Infrastructure.VS/GitWorkSpaceService.cs

```diff
@@ -0,0 +1,86 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2022 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.ComponentModel.Composition;
+using System.IO;
+using System.IO.Abstractions;
+using SonarLint.VisualStudio.Integration;
+
+namespace SonarLint.VisualStudio.Infrastructure.VS
```

Comment:
Does this need to be in the `Infrastructure.VS` project?
It doesn't use or expose any types that are VS-specific.

Reply:
I couldn't think of a better place to put it. I don't think it would make sense to put it in the new assembly because this can be used for anything. And it relies on "IWorkspaceService" which is VS-specific.

Reply:
Hhmm. The implementations of both `IWorkspaceService` and `IFolderWorkspaceService` are dependant on VS types/services, but the interfaces don't expose any VS types so they could be in `Core`.

I think we should put all of the interfaces into `Core` since they don't depend on VS (and it increases where they can be used).

I think I would also put the implementation of this new service in `Core` too, because it's implementation doesn't have any dependency on VS.